src/audio/Audio.js

```diff
@@ -58,23 +62,19 @@ THREE.Audio.prototype.setNodeSource = function ( audioNode ) {
 
 };
 
-THREE.Audio.prototype.setBuffer = function ( audioBuffer ) {
+THREE.Audio.prototype.setBuffer = function( audioBuffer ) {
 
 	var scope = this;
 
-	audioBuffer.onReady( function( buffer ) {
-
-		scope.source.buffer = buffer;
-		scope.sourceType = 'buffer';
-		if ( scope.autoplay ) scope.play();
-
-	} );
+	scope.source.buffer = audioBuffer;
+	scope.sourceType = 'buffer';
+	if ( scope.autoplay ) scope.play();
 
 	return this;
 
 };
 
-THREE.Audio.prototype.play = function () {
```

Comment:
Seems like you removed the space in `function () {` in a bunch of places. Do you mind reverting that?